data/binding/printf_test.go

```diff
@@ -0,0 +1,108 @@
+package binding
+
+import (
+	"fmt"
+	"testing"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSprintfConversionRead(t *testing.T) {
+	var b bool = true
+	var f float64 = 42
+	var i int = 7
+	var r rune = 'B'
+	var s string = "this is a string"
+	var u fyne.URI = storage.NewFileURI("/")
+
+	bb := BindBool(&b)
+	bf := BindFloat(&f)
+	bi := BindInt(&i)
+	br := BindRune(&r)
+	bs := BindString(&s)
+	bu := BindURI(&u)
+
+	format := "Bool %v, Float %f, Int %i, Rune %v, String '%s', URI '%s'"
+	sp, err := NewSprintf(format, bb, bf, bi, br, bs, bu)
+	expected := fmt.Sprintf(format, b, f, i, r, s, u)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, sp)
+
+	waitForItems()
+
+	sGenerated, err := sp.Get()
+
+	assert.Nil(t, err)
+	assert.NotNil(t, sGenerated)
+	assert.Equal(t, expected, sGenerated)
+
+	bb.Set(false)
+	bf.Set(7)
+	bi.Set(42)
+	br.Set('C')
+	bs.Set("a different string")
+
+	expectedChange := fmt.Sprintf(format, b, f, i, r, s, u)
+
+	waitForItems()
+
+	sChange, err := sp.Get()
+
+	assert.Nil(t, err)
+	assert.NotNil(t, sChange)
+	assert.Equal(t, expectedChange, sChange)
+	assert.NotEqual(t, sGenerated, sChange)
+}
+
+func TestSprintfConversionReadWrite(t *testing.T) {
+	var b bool = true
+	var f float64 = 42
+	var i int = 7
+	var r rune = 'B'
+	var s string = "this is a string"
+
+	bb := BindBool(&b)
+	bf := BindFloat(&f)
+	bi := BindInt(&i)
+	br := BindRune(&r)
+	bs := BindString(&s)
+
+	format := "Bool %v , Float %f , Int %v , Rune %v , String %s"
+	sp, err := NewSprintf(format, bb, bf, bi, br, bs)
+	expected := fmt.Sprintf(format, b, f, i, r, s)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, sp)
+
+	waitForItems()
+
+	sGenerated, err := sp.Get()
+
+	assert.Nil(t, err)
+	assert.NotNil(t, sGenerated)
+	assert.Equal(t, expected, sGenerated)
+
+	err = sp.Set("Bool false , Float 7.000000 , Int 42 , Rune 67 , String nospacestring")
```

Comment:
I think we should check strings with spaces as well, user data will likely have some :)

Reply:
It is actually a limitation of scanf. I didn't feel like trying to give it a different behavior as that could be more confusing.

Reply:
I get that Scanf has some limitations, but this is a binding for SPrintf in a 2 way mode - will people be expecting the Scanf limitations to apply?
If this is indeed to be expected I think we need to be clear that it will not work if strings have spaces - possibly capturing it as an error in our binding - I am assuming that otherwise it will crash or behave unexpectecly?

Reply:
It won't crash, it just will parse things in unexpected way or refuse to parse string. Go scanf rely on spaces to split the string given and extract each component. This means that it is not just string containing spaces, but even the way the format is specified that require to take spaces into account.

We could change the name or/and improve documentation to make it clear that this two way binding will have scanf limitation when used with for setting data.